feat(api): add pagination to license browser

mayank-pathakk opened the pull request:
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only
-->

<!-- Please refer to CONTRIBUTING.md (https://github.com/fossology/fossology/blob/master/CONTRIBUTING.md)
before creating the pull request to make sure you follow all the standards. -->

# Description

As mentioned in the issue #2404, pagination feature needed to be added to the endpoint "uploads/{id}/license".
Hence, this PR aims to solve the same. 

## Changes

Additional fields ( i.e. limits and page ) can be passed to UploadController, which calls for UploadHelper. While receiving response from UploadHelper, the response array is modified using `array_slice` according to the needed page and limit fields. 

Also updated the `open-api.yaml` to display the changes.



<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2430"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>


## Testing
1.) Using an auth token send a get request to `uploads/{id}/license` . This will display all the results.
2.) Include page and limit in the headers of the request. This will display the paginated results.




### Results

### Before :

![Screenshot from 2023-04-06 23-31-05](https://user-images.githubusercontent.com/97532378/230462624-4a848859-3bb6-44eb-bc7f-4a68ae4c81ec.png)

### After : 

![Screenshot from 2023-04-06 23-31-18](https://user-images.githubusercontent.com/97532378/230462724-f22f7f46-7858-4968-9377-226d73eea118.png)

GMishx commented on src/www/ui/api/documentation/openapi.yaml:
```diff
@@ -654,6 +654,23 @@ paths:
+            type: integer
+            default: 50
+            minimum: 1
+            maximum: 100
```
For this particular endpoint, I think the max limit can be increased to 1000

mayank-pathakk replied:
Corrected it

GMishx commented on src/www/ui/api/Helper/UploadHelper.php:
```diff
@@ -821,6 +821,9 @@ public function getUploadLicenseList($uploadId, $agents, $printContainers, $bool
       }
     }
-    return $responseList;
+    $Offset = $Page * $Limit;
```
Please use proper naming convention for variables. They should follow camel casing.

GMishx commented on src/www/ui/api/Helper/UploadHelper.php:
```diff
@@ -821,6 +821,9 @@ public function getUploadLicenseList($uploadId, $agents, $printContainers, $bool
-    return $responseList;
+    $Offset = $Page * $Limit;
+    $paginatedResponseList = array_slice($responseList, $Offset, $Limit);
+    $paginatedResponseListSize = sizeof($paginatedResponseList);
```
`$paginatedResponseListSize` should represent the total size of array, not the size of response. Otherwise client will never know how many pages it can fetch.

mayank-pathakk commented:
@GMishx sir I have made the requested changes, rebased my PR, and fixed the tests, which were causing 2 workflows to fail, now all the workflows have passed. :)

dushimsam commented:
These changes look good to me. Related issue [#2453 ](https://github.com/fossology/fossology/issues/2453)

@GMishx  @shaheemazmalmmd

mayank-pathakk commented:
Rebased

GMishx reviewed (approved):
Changes looks good.
Tested, working as expected.
